src/Native/System.Security.Cryptography.Native/pal_cms.h

```diff
@@ -63,47 +63,52 @@ Shims the CMS_RecipientInfo_free method
 extern "C" void CryptoNative_CmsRecipientDestroy(CMS_RecipientInfo* recipient);
 
 /*
-Initializes a CMS_ContentInfo with the cipher given by the oid in algorithmOid and returns a pointer
-to it. 
+Initializes a CMS_ContentInfo of type EnvelopedData with the cipher given by the oid in algorithmOid
+and returns a pointer to it. 
 
-Status is set to 1 on success, 0 on unsupported algorithm, and -1 on null input.
+Status is set to 1 on success, 0 on unsupported algorithm, and -1 on invalid input.
 */
-extern "C" CMS_ContentInfo* CryptoNative_CmsEncryptInitialize(ASN1_OBJECT* algorithmOid, int32_t* status);
+extern "C" CMS_ContentInfo* CryptoNative_CmsInitializeEnvelope(ASN1_OBJECT* algorithmOid, int32_t* status);
 
 /*
-Adds all the originator certificates to the CMS_ContentInfo structure.
+Adds an originator certificate to the CMS_ContentInfo structure. If the CMS_ContentInfo structure
+is of type EnvelopedData it will be added to OriginatorInfo, and for SignedData it will be added to
+the certificates.
 
-Returns 1 on success, 0 on OpenSSL failure, and -1 on null input.
+Returns 1 on success, 0 on OpenSSL failure, and -1 on invalid input.
 */
-extern "C" int CryptoNative_CmsAddOriginatorCerts(CMS_ContentInfo* cms, X509Stack* originatorCerts);
+extern "C" int CryptoNative_CmsAddCert(CMS_ContentInfo* cms, X509Stack* originatorCerts);
```

Comment:
I don't see how this is satisfied.  The implementation is CryptoNative_CmsAddOriginatorCert, which is a better name.  And takes a single cert, instead of a stack.